molecule-runtime/src/commonMain/kotlin/app/cash/molecule/GatedFrameClock.kt

```diff
@@ -46,8 +46,23 @@ internal class GatedFrameClock(scope: CoroutineScope) : MonotonicFrameClock {
       }
     }
 
+  private var lastNanos = 0L
+  private var lastOffset = 0
```

Comment:
Is there any reason to not just use a monotonically incrementing counter if we're going to +1 on a collision?

Reply:
There's only 1,000,000 nanos in a second but 4,000,000,000 ints. Actually 2,000,000,000 and then we roll over to -2,000,000,000. This basically means we skew the real time further and further until we jump backwards (which also means you could observe the clock going backwards).

It sucks keeping both, but I can't see a path to anything simpler. I tried a bunch of things.